### PR TITLE
Event Publishing

### LINKED ISSUE #9: Event publishing
Lifecycle manager should publish kubernetes events for the following:

- Lifecycle Hook Received
- Target Instance Deregistered from TargetGroup
- Node Drained
- Lifecycle Hook Processed Successfully / Failed

### PR DESCRIPTION
**Pull request by eytan-avisror**
Fixes #9 

This enables publishing of Kubernetes events for lifecycle hooks

## Testing
Added unit tests and did a full test with large termination activities.

### CODE REVIEW

File: pkg/service/events.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	EventMessageTargetDeregisterSucceeded             = "target %v:%v has successfully deregistered from target group %v"
+
+	EventReasonTargetDeregisterFailed  EventReason = "EventReasonTargetDeregisterFailed"
+	EventMessageTargetDeregisterFailed             = "target %v:%v has failed to deregistered from target group %v: %v"
+
+	EventReasonInstanceDeregisterSucceeded  EventReason = "EventReasonInstanceDeregisterSucceeded"
+	EventMessageInstanceDeregisterSucceeded             = "instance %v has successfully deregistered from classic-elb %v"
+
+	EventReasonInstanceDeregisterFailed  EventReason = "EventReasonInstanceDeregisterFailed"
+	EventMessageInstanceDeregisterFailed             = "instance %v has failed to deregistered from classic-elb %v: %v"
```
Comment (mattouille) [inline pkg/service/events.go]: Are these `EventReason`'s good candidates for constants?

Comment (eytan-avisror) [reply pkg/service/events.go]: Yes, I don't see anything or any test that would need to change these.

File: pkg/service/server.go [hunk 2]
```diff
@@ -192,10 +199,11 @@ func (mgr *Manager) newPoller() {
 
 func (mgr *Manager) newWorker(message *sqs.Message) {
```
Comment (mattouille) [inline pkg/service/server.go]: This doesn't look like it needs a pointer receiver

Comment (eytan-avisror) [reply pkg/service/server.go]: This is out of bound for this PR as it wasn't added here.
I'll take a look but pretty sure I need to work on a pointer receiver here due to downstream mutations of `ManagerContext`.

**Pull request merged**